src/prediction.jl

```diff
@@ -249,4 +249,68 @@ function build_future_df(base_df, to_predict::Symbol)
 
     return(reduce(vcat, result_lines))
 
-end # end function1
\ No newline at end of file
+end # end function1
+
+function tryparsecol(Type::DataType, col)
+    try
+        return(parse.(Type, col))
+    catch
+        return(col)
+    end
+end
+
+function univariate_tietjenmoore(values::Vector{T} where T <: Real, k::Int64; alpha = 0.05, sim_trials = 100000, return_indexes = true)
+
+    function compute_tietjenmoore(data, k, n)
+        r_all = abs.(data .- mean(data))
+        filteredData = data[sortperm(r_all)][1:(n-k)]
+        ksub = (filteredData .- mean(filteredData))
+    
+        return( sum(ksub .^ 2) / sum(r_all .^ 2) )
+    end
+    
+    function test_tietjenmoore(dataSeries,k, n, alpha, sim_trials)
+        ek = compute_tietjenmoore(dataSeries,k, n)
+        simulation = [ compute_tietjenmoore(randn(length(dataSeries)), k, n) for i in 1:sim_trials ]
+        Talpha=quantile(simulation,alpha)
+
+        return(ek, Talpha)
+    end
+
+    println("----- Begin Tietjen-Moore Outlier test -----")
```

Comment:
Let's use logging rather than printing inside functions so that (in principle) stuff can be silenced if we want.

Reply:
```suggestion
    @info "----- Begin Tietjen-Moore Outlier test -----"
```